### PR TITLE
[Pricing] Price per channel and currency

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| License         | MIT |

### CODE REVIEW

File: features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        And the store operates on a channel named "Web"
+        And the store has a "Wyborowa Vodka" configurable product
+        And this product has option "Taste" with values "Orange" and "Melon"
+        And I am logged in as an administrator
+
+    @ui @javascript
+    Scenario: Configure channel and currency calculator while adding new simple product
+        Given I want to create a new variant of this product
+        When I specify its code as "VODKA_WYBOROWA_PREMIUM"
+        And I set its price to "$100.00"
```
Comment (michalmarcinkowski) [inline features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature]: maybe we can say here `And I set its default price to "$100.00"`?

File: features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    Background:
+        Given the store is available in "English (United States)"
+        And the store operates on a channel named "Web"
+        And the store has a "Wyborowa Vodka" configurable product
+        And this product has option "Taste" with values "Orange" and "Melon"
+        And I am logged in as an administrator
+
+    @ui @javascript
+    Scenario: Configure channel and currency calculator while adding new simple product
```
Comment (michalmarcinkowski) [inline features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature]: new ~~simple product~~ variant

File: src/Sylius/Bundle/CoreBundle/Form/DataTransformer/ChannelAndCurrencyPricingConfigurationTransformer.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace Sylius\Bundle\CoreBundle\Form\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class ChannelAndCurrencyPricingConfigurationTransformer implements DataTransformerInterface
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Form/DataTransformer/ChannelAndCurrencyPricingConfigurationTransformer.php]: IMO it should be a part of `ChannelAndCurrencyBasedConfigurationType`, it's not gonna be used anywhere else.

File: features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature [hunk 4]
```diff
@@ -0,0 +1,23 @@
+@managing_product_variants
+Feature: Configure channel and currency calculator
```
Comment (GSadee) [inline features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature]: I guess that we have convention for naming feature files with `-ing`, so I guess it should be `Configuring...`

File: features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    Background:
+        Given the store is available in "English (United States)"
+        And the store operates on a channel named "Web"
+        And the store has a "Wyborowa Vodka" configurable product
+        And this product has option "Taste" with values "Orange" and "Melon"
+        And I am logged in as an administrator
+
+    @ui @javascript
+    Scenario: Configure channel and currency calculator while adding new variant
+        Given I want to create a new variant of this product
```
Comment (GSadee) [inline features/product/managing_product_variants/configure_channel_and_currency_calculator_for_product_variant.feature]: `When I want...`

File: features/product/managing_products/configure_channel_and_currency_calculator_for_product.feature [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    I want to add a new product to the shop with different prices
+
+    Background:
+        Given the store is available in "English (United States)"
+        And the store operates on a channel named "Web"
+        And I am logged in as an administrator
+
+    @ui @javascript
+    Scenario: Configure channel and currency calculator while adding new simple product
+        Given I want to create a new simple product
```
Comment (GSadee) [inline features/product/managing_products/configure_channel_and_currency_calculator_for_product.feature]: `When`

File: src/Sylius/Behat/Page/Admin/Product/UpdateSimpleProductPage.php [hunk 7]
```diff
@@ -286,6 +288,16 @@ public function removeAssociatedProduct($productName, ProductAssociationTypeInte
     /**
      * {@inheritdoc}
      */
+    public function getPricingConfigurationForChannelAndCurrencyCalculator(ChannelInterface $channel, CurrencyInterface $currency)
+    {
+        $priceElement = $this->getElement('pricing_configuration')->find('css', sprintf('label:contains("%s %s")', $channel->getCode(), $currency->getCode()))->getParent();
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/Product/UpdateSimpleProductPage.php]: Too long line I guess

**Comment by pjedrzejewski** [comment]
Thank you Arek, please apply @GSadee's comments in separate PR.

**Pull request merged**